flocx_market/db/sqlalchemy/api.py

```diff
@@ -39,19 +40,34 @@ def drop_db():
 
 
 def offer_get(marketplace_offer_id, context):
-
-    return get_session().query(models.Offer).filter_by(
+    offer_ref = get_session().query(models.Offer).filter_by(
             marketplace_offer_id=marketplace_offer_id).first()
 
+    if offer_ref:
+        return offer_ref
+    else:
+        raise exception.ResourceNotFound(resource_type="Offer",
+                                         resource_uuid=marketplace_offer_id)
+
 
 def offer_get_all(context):
-    return get_session().query(models.Offer).all()
+    offer_refs = get_session().query(models.Offer).all()
+
+    if offer_refs:
+        return offer_refs
+    else:
+        raise exception.ResourceQueryNotFound(resource_type="Offer")
 
 
 def offer_get_all_by_project_id(context):
-    return get_session().query(models.Offer).filter_by(
+    offer_refs = get_session().query(models.Offer).filter_by(
         project_id=context.project_id).all()
 
+    if offer_refs:
+        return offer_refs
+    else:
+        return None
```

Comment:
I think you could just return offer_refs, which would return an empty list in the case that it didn't match anything, right?